tests/test_query.py

```diff
@@ -0,0 +1,205 @@
+import httpx
+import pytest
+from fastapi import status
+
+from app.api import utility as util
+
+
+@pytest.fixture()
+def test_single_matching_dataset_result():
```

Comment:
since this is a fixture, maybe we can call it 
```suggestion
def mocked_single_matching_dataset_result():
```
I'm not sure if pytest will ignore it despite starting with `test_`, but probably easier for humans to make sense of